### ISSUE TITLE
Release v1.4.0

### ISSUE THREAD

**Issue by AirQuick** [xspec/xspec#594]
[Project board](https://github.com/xspec/xspec/projects/5) is mostly done. The remaining cards will be resolved in a week or two:

* #475
    This will be the highlight of v1.4.0. I'm actively reviewing the pull request now and have reviewed 27/40 files.
* #593
  This may be postponed to a later release depending on how it develops.

So, unless @cirulls and @galtm are entering their busy season, I would like to propose releasing v1.4.0 within two or three weeks.

Task list taken from [Release Workflow](https://github.com/xspec/xspec/wiki/Release-Workflow):

- [x] Write release notes
- [x] Create a pull request incrementing the version of pom.xml and merge it into master (like #567)
- [x] Deploy the XSpec artifact to Maven Central following [these instructions](https://github.com/xspec/xspec/wiki/Deploy-XSpec-artifact-to-Maven-Central)
- [x] Publish the release
- [x] Close the project related to the new release
- [x] Promote the release
- [x] Create a project for the new release
- [x] Create a pull request adding SNAPSHOT in pom.xml and merge it into master (like #568)

**Comment by galtm** [comment]
The time frame sounds good to me. If your continue to enter your code review comments for #475 gradually, I will do my best to keep up.

**Comment by cirulls** [comment]
This sounds good to me, thanks for the early warning. I'm happy to do the usual release tasks once the code is ready. Looking forward to XSpec v1.4.0!

**Comment by AirQuick** [comment]
Thanks @cirulls 
This time, will you be able to try this task?
> Deploy the XSpec artifact to Maven Central following these instructions

**Comment by cirulls** [comment]
Yes, I can give it a go. @cmarchand: I may need your help in case I fail but I'll try first.

**Comment by cmarchand** [comment]
Ok, @cirulls , but do not delay asking help...
Best

**Comment by cirulls** [comment]
Thanks @cmarchand, I'll give it a go at release time (in 2-3 weeks) and will keep you posted.

**Comment by AirQuick** [comment]
I think the highlight would include these:
* XQuery support is mostly on a parity with XSLT. Now implementing new XQuery features is easier than ever. Any contribution is welcome.
* Test coverage continues growing. We have [more than 200 tests](https://ci.appveyor.com/project/xspec/xspec/builds/27047775) running on our CI systems.

**Comment by cirulls** [comment]
I added these highlights to the draft release notes. I should be able to release XSpec v1.4.0 on Sunday 8th September.

**Comment by cirulls** [comment]
An update on release v1.4.0:

- I created a pull request incrementing the version of pom.xml and merge it into master (#614).
- I deployed the XSpec artifact to Maven Central. I can see the artifact under [repo1.maven.org](https://repo1.maven.org/maven2/io/xspec/xspec/1.4.0/). I was expecting to see the artifact also via [search.maven.org](https://search.maven.org/search?q=g:io.xspec%20AND%20a:xspec&core=gav) but it has not appeared yet. I don't know if it is a matter of time (I had to wait few minutes before seeing the artifact under repo1.maven.org). @cmarchand: do I know if have to do something else to see the artifact under search.maven.org?
- I published the release via [GitHub releases](https://github.com/xspec/xspec/releases/tag/v1.4.0). 

I'm waiting for AppVeyor to complete before promoting the release via our usual channels.

A note on the release process: as the `pom.xml` is already set up and as I already have  `~/.m2/settings.xml` correctly set up on my local machine, all I had to do was to issue the command `mvn -Prelease deploy`. A large part [these instructions](https://github.com/xspec/xspec/wiki/Deploy-XSpec-artifact-to-Maven-Central) refers to how to set it up for the first time (which @cmarchand nicely did it for XSpec). Once `~/.m2/settings.xml` is correctly set up with username and password for xspec, the deployment to Maven is just a matter of issuing that command from the root of xspec repository. I will clarify this to the documentation.

**Comment by cmarchand** [comment]
You have to wait many days - 4 to 10 - to see your new release on search.maven.org. I do not understand how the update their index database, it's a mistery. But if you wait enough, it appears !

**Comment by cirulls** [comment]
@cmarchand: it looks like this time they were quicker, the artifact for v1.4.0 is now visible on search.maven.org, see [here](https://search.maven.org/artifact/io.xspec/xspec/1.4.0/jar). 

I promoted the new release on [xml.com](https://www.xml.com/news/2019-09-xspec-v140) and on the [XSpec Google Groups](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/xspec-users/3K3LPQs491M/jVOXA072AwAJ). I also sent an email to the xml-dev mailing list. However, despite I regularly receive the email from the xml-dev mailing list, [my email doesn't seem to get published](http://lists.xml.org/archives/xml-dev/201909/maillist.html) on the mailing list. I contacted their technical support to see what I have to do. 

@cmarchand and @yamahito: I know you are active on social media, could you please tweet about the new XSpec release when you have time? Thanks!

**Comment by cirulls** [comment]
I raised #617 to update the documentation and submitted #616 to add SNAPSHOT in the pom file. 

I think we can close this issue now that v1.4.0 is out. If anyone spots anything missing from the release process, feel free to reopen the issue or create a new one. 

Many thanks to all those who contributed to this release!!!

**Comment by AirQuick** [comment]
Thanks @cirulls for the smooth operation.

**Comment by AirQuick** [comment]
@AlexJitianu 
You may possibly want to update XSpec Framework for XSpec Helper view.
Also, looking at [the repository](https://github.com/xspec/oXygen-XML-editor-xspec-support/tree/master/frameworks/xspec/oxygen-results-view), I noticed you needed to patch XSpec compiler. Would it be worth integrating some of the patches into this main repository?

**Comment by cmarchand** [comment]
@cirulls : promoted on xml-dev and xsl-list mailing lists, and on twitter.
 - https://xsl.markmail.org/search/?q=XSpec%201.4.0#query:XSpec%201.4.0+page:1+mid:7s6orbte4zacg2ev+state:results
 - https://xml-dev.markmail.org/search/?q=XSpec%201.4.0#query:XSpec%201.4.0+page:1+mid:yts4vwxatth25xzo+state:results
 - https://twitter.com/cmarchand7276/status/1170959298949783552
